cmd/scaffold.go

```diff
@@ -167,8 +167,26 @@ func (fnb *FlowNodeBuilder) enqueueNetworkInit() {
 			myAddr = fnb.BaseConfig.bindAddr
 		}
 
-		mw, err := p2p.NewMiddleware(fnb.Logger.Level(zerolog.ErrorLevel), codec, myAddr, fnb.Me.NodeID(),
-			fnb.networkKey, fnb.Metrics.Network, p2p.DefaultMaxUnicastMsgSize, p2p.DefaultMaxPubSubMsgSize,
+		libP2PNodeFactory, err := p2p.DefaultLibP2PNodeFactory(fnb.Logger.Level(zerolog.ErrorLevel),
+			fnb.Me.NodeID(),
+			fnb.Me.Address(),
+			fnb.networkKey,
+			fnb.RootBlock.ID().String(),
+			p2p.DefaultMaxPubSubMsgSize,
+			fnb.Metrics.Network)
+		if err != nil {
+			return nil, fmt.Errorf("could not generate libp2p node factory: %w", err)
+		}
+
+		mw, err := p2p.NewMiddleware(fnb.Logger.Level(zerolog.ErrorLevel),
```

Comment:
some of these params are no longer needed to be passed in to `NewMiddleware` since they are already passed in to the `libP2PNodeFactory`.
1. `RootBlockID`
2. `myAddr` may be skipped as it is set on `middleware.Start` sets the `host` and `port`.
3. `fnb.networkKey` since there is no caller for `middleware.PublicKey()`

Reply:
fixed in https://github.com/onflow/flow-go/pull/195/commits/95c5512b0e4ed9123e7e046544d560655c9eb285 and https://github.com/onflow/flow-go/pull/195/commits/dd3bb4a3778f1f6c6cdeedb0721cff84672be9d1.